### PR TITLE
Bug fix: New endpoints in cloud config should be saved in config for the known clouds

### PR DESCRIPTION
**Pull request by derekbekoe**
Closes https://github.com/Azure/azure-cli/issues/2905

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [n/a] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [n/a] Each command and parameter has a meaningful description.
- [n/a] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by derekbekoe** [comment]
fyi @annatisch. This PR ensures that the endpoints set in code are added to the config file if they are missing. This should handle new endpoints that get added after the config has already been created.

**Comment by annatisch** [comment]
@derekbekoe - that's awesome, thank you!! :)

File: src/azure-cli-core/azure/cli/core/cloud.py [hunk 1]
```diff
@@ -230,18 +230,18 @@ def get_custom_clouds():
     return [c for c in get_clouds() if c.name not in known_cloud_names]
 
 
-def _init_known_clouds():
+def init_known_clouds(force=True):
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/cloud.py]: Do you mean to have the default value at `True`? Callers are not passing in `False`, so to me, this arg is not really needed.

Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/cloud.py]: Yes it should have been `False`. fixed.

**Pull request merged**